sql/src/test/java/io/crate/operation/join/NestedLoopOperationTest.java

```diff
@@ -290,6 +292,134 @@ public void testFailAndFinishResultsInFailure() throws Exception {
     }
 
     @Test
+    public void testRightUpstreamFailureOnLeftFirstResultsInFailure() throws Exception {
+        CollectingRowReceiver receiver = new CollectingRowReceiver();
+
+        NestedLoopOperation nestedLoopOperation = unfilteredNestedLoopOperation(0, receiver);
+        ListenableRowReceiver left = nestedLoopOperation.leftRowReceiver();
+        ListenableRowReceiver right = nestedLoopOperation.rightRowReceiver();
+
+        List<Row> leftRows = singleColRows(1, 2, 3);
+        RowSender rsLeft = new RowSender(leftRows, left, executorService);
+        RowSender rsRight = RowSender.withFailure(right, executorService); // fail immediately
+        executorService.submit(rsLeft);
+        Thread.sleep(1);
```

Comment:
I assume the sleep is here to make sure the left row is submitted before the right? Would it be possible to just call rsLeft.run() within the test synchronous or does that behave differently?